### ISSUE TITLE
Package name doesn't match repository name

### ISSUE THREAD

**Issue by crocodile2u** [clue/php-block-react#64]
I went into composer error today:

```
$ composer require clue/reactphp-block
$ [InvalidArgumentException]                                                                                                                                                                                      
  Could not find a matching version of package clue/reactphp-block
...
```

Took a while before I noticed that the composer package name is `clue/block-react`, not `clue/reactphp-block`. Also, on https://reactphp.org/http/#client-usage only `clue/reactphp-block` is mentioned.

Proposed fix: add installation instructions to README. If possible, mention correct package on reactphp website.

**Comment by clue** [comment]
@crocodile2u Thanks for reporting, this is an interesting one!

The [install instructions](https://github.com/clue/reactphp-block#install) in the documentation show the correct command and package name to use:

```php
$ composer require clue/block-react
```

I understand that this can be slightly confusing and would ideally be avoided. To be clear, this project is called "clue/reactphp-block", but the name used for Composer/Packagist is "clue/block-react" for historical reasons. All of my newer packages are consistently named on GitHub and Composer/Packagist, but for my older packages such as this one, there's still this minor inconsistency.

I've considered renaming the packages on Composer/Packagist before, but unfortunately there's only poor tool support for this use case. Renaming this package now would essentially mean submitting a *new* package under the name "clue/reactphp-block" and then marking the *old* package "clue/block-react" as abandoned. This would allow redirecting users to the new package name and would require some manual effort for every consumer of this package to update their `composer.json`. Additionally, this would reset download counts to zero for the new package. While this is not something I would be too worried about, I also understand that some people might refer to this as a metric to see how relevant and/or reliable a given project is (*old* currently has 822k, *new* would be 0 initially).

To sum this up, I'm undecided if it's worth the effort and if the pros outweight the cons. Would love to hear more feedback on this 👍

**Comment by crocodile2u** [comment]
Thanks for the quick answer, @clue . I didn't suggest to change anything - not the repo name and not the package name. I understand there's some history behind it. It's going to be better if this mismatch is somehow mentioned in the docs though. Adding installation instructions which would point to a correct package name, would be enough))

**Comment by clue** [comment]
@crocodile2u Again thank you for bringing this up!

I think this is an interesting discussion, so I've also asked this question on Twitter and the overwhelming consensus seems to be that this inconsistency (while certainly not desirable) is acceptable: https://twitter.com/another_clue/status/1454397008748421124

Given your reply and the fact the README already has some [installation instructions](https://github.com/clue/reactphp-block#install), I think this has been answered and can be closed. If you feel there's anything that needs to be done here, please let me know and we can always reopen this.

On top this, I hear that (parts of) this package will soon live under a different name anyway, but 🤫 (https://github.com/reactphp/async/pull/8)

Thank you!
